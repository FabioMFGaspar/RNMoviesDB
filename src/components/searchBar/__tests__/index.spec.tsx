import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import SearchBar from '..';

describe('SearchBar test suite', () => {
  const mockHandle = jest.fn();

  it('Default snapshot test', () => {
    const {toJSON} = render(
      <SearchBar query="dummyQuery" handleSearch={mockHandle} />,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('Should trigger onChangeText', () => {
    const {getByPlaceholderText} = render(
      <SearchBar query="dummyQuery" handleSearch={mockHandle} />,
    );

    fireEvent(getByPlaceholderText('Search...'), 'onChangeText', 'dummyChange');
    expect(mockHandle).toHaveBeenCalledWith('dummyChange');
  });
});
