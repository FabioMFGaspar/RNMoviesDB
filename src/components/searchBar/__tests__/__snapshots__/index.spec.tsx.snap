// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar test suite Default snapshot test 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#000033",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#212145",
        "borderRadius": 5,
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginHorizontal": 15,
        "marginVertical": 10,
        "padding": 10,
      }
    }
  >
    <TextInput
      onChangeText={[MockFunction]}
      placeholder="Search..."
      placeholderTextColor="white"
      style={
        Object {
          "backgroundColor": "#212145",
          "color": "white",
          "flex": 1,
          "padding": 0,
        }
      }
      value="dummyQuery"
    />
    <Image
      source={
        Object {
          "uri": "https://img.icons8.com/ios-glyphs/30/000000/search--v2.png",
        }
      }
      style={
        Object {
          "height": 25,
          "marginLeft": 5,
          "tintColor": "white",
          "width": 25,
        }
      }
    />
  </View>
</View>
`;
